tests/test_bcrypt.py

```diff
@@ -494,3 +498,41 @@ def test_2a_wraparound_bug():
         )
         == b"$2a$04$R1lJ2gkNaoPGdafE.H.16.1MKHPvmKwryeulRe225LKProWYwt9Oi"
     )
+
+
+@pytest.mark.thread_unsafe()
+def test_multithreading():
+    def get_id():
+        return uuid.uuid4().bytes
+
+    class User:
+        def __init__(self, id_, pw):
+            self.id_ = id_
+            self.salt = bcrypt.gensalt(4)
+            self.hash_ = bcrypt.hashpw(pw, self.salt)
+            self.key = bcrypt.kdf(pw, self.salt, 32, 50)
+            assert self.check(pw)
+
+        def check(self, pw):
+            return bcrypt.checkpw(pw, self.hash_)
+
+    # use UUIDs as both ID and passwords
+    num_users = 50
+    ids = [get_id() for _ in range(num_users)]
+    pws = {id_: get_id() for id_, _ in zip(ids, range(num_users))}
+
+    user_creator = ThreadPoolExecutor(max_workers=4)
+
+    def create_user(id_, pw):
+        return id_, User(id_, pw)
+
+    creator_futures = [
+        user_creator.submit(create_user, id_, pw) for id_, pw in pws.items()
+    ]
+
+    users = [future.result() for future in creator_futures]
+
```

Comment:
I feel like there's way too much going on relative to what we're _actually_ trying to test. At base we're trying to to run all the bcrypt functions concurrently.

So I think this can simplify to:

```
futures = [user_creator.submit(create_user, uuid.uuid4()) for _ in range(num_users)]
```

remoe the unused id field from User, and then just use the attributes, rather than the pws dict?